universe/System.cpp

```diff
@@ -328,42 +330,49 @@ void System::Insert(TemporaryPtr<UniverseObject> obj, int orbit/* = -1*/) {
     obj->SetSystem(this->ID());
 
     if (obj->ObjectType() == OBJ_PLANET) {
+        int new_orbit = orbit;
         if (orbit == -1) {
-            bool already_in_orbit = false;
             for (int o = 0; o < static_cast<int>(m_orbits.size()); ++o) {
-                if (m_orbits[o] = obj->ID()) {
-                    already_in_orbit = true;
+                if (m_orbits[o] == obj->ID()) {
+                    new_orbit = o;
                     break;
                 }
             }
-            // if planet already in an orbit, do nothing
-            // if planet not in an orbit, find orbit to put planet in
-            if (!already_in_orbit) {
-                for (int o = 0; o < static_cast<int>(m_orbits.size()); ++o) {
-                    if (m_orbits[o] == INVALID_OBJECT_ID) {
-                        orbit = o;
-                        m_orbits[orbit] = obj->ID();
-                        break;
-                    }
+            if (new_orbit == -1) {
+                new_orbit = RandomFreeOrbit();
+                if (new_orbit == -1) {
+                    ErrorLogger() << "No free orbits in system " << ID()
+                                  << " to place planet " << obj->ID();
+                    return;
                 }
             }
         } else {
-            // check if desired orbit is occupied
-            // if yes, it is either already occupied by the inserted object or
-            // by another object. in either case, do nothing.
-            // if not, put object into the orbit, and remove from any other
-            // orbit it currently occupies.
-            if (!OrbitOccupied(orbit)) {
+            // planet given a specific oribt, check if already assigned
```

Comment:
My fingers take exception to this, apparently determined that this is the correct spelling.  Thank you, will fix.